osquery/tables/system/windows/logical_drives.cpp

```diff
@@ -18,78 +18,37 @@ QueryData genLogicalDrives(QueryContext& context) {
   QueryData results;
 
   const WmiRequest wmiLogicalDiskReq(
-      "select DeviceID, DriveType, FreeSpace, Size, FileSystem from "
+      "select DeviceID, Description, FreeSpace, Size, FileSystem from "
       "Win32_LogicalDisk");
-  const std::vector<WmiResultItem>& wmiResults = wmiLogicalDiskReq.results();
-  for (unsigned int i = 0; i < wmiResults.size(); ++i) {
+  const std::vector<WmiResultItem>& logicalDisks = wmiLogicalDiskReq.results();
+  const WmiRequest wmiBootConfigurationReq(
+      "select BootDirectory from Win32_BootConfiguration");
+  const std::vector<WmiResultItem>& bootConfigurations =
+      wmiBootConfigurationReq.results();
+
+  for (const auto& logicalDisk : logicalDisks) {
     Row r;
-    unsigned int driveType = 0;
     std::string deviceId;
-    wmiResults[i].GetString("DeviceID", deviceId);
-    wmiResults[i].GetUnsignedInt32("DriveType", driveType);
-    wmiResults[i].GetString("FreeSpace", r["free_space"]);
-    wmiResults[i].GetString("Size", r["size"]);
-    wmiResults[i].GetString("FileSystem", r["file_system"]);
-
-    r["device_id"] = deviceId;
-
-    switch (driveType) {
-    default:
-      r["type"] = TEXT("Unknown");
-      break;
-    case 1:
-      r["type"] = TEXT("No Root Directory");
-      break;
-    case 2:
-      r["type"] = TEXT("Removable Disk");
-      break;
-    case 3:
-      r["type"] = TEXT("Local Disk");
-      break;
-    case 4:
-      r["type"] = TEXT("Network Drive");
-      break;
-    case 5:
-      r["type"] = TEXT("Compact Disc");
-      break;
-    case 6:
-      r["type"] = TEXT("RAM Disk");
-      break;
-    }
-
-    r["boot_partition"] = INTEGER(0);
-
-    std::string assocQuery =
-        std::string("Associators of {Win32_LogicalDisk.DeviceID='") + deviceId +
-        "'} where AssocClass=Win32_LogicalDiskToPartition";
-
-    const WmiRequest wmiLogicalDiskToPartitionReq(assocQuery);
-    const std::vector<WmiResultItem>& wmiLogicalDiskToPartitionResults =
-        wmiLogicalDiskToPartitionReq.results();
-
-    if (wmiLogicalDiskToPartitionResults.empty()) {
-      results.push_back(r);
-      continue;
+    int bootPartition = 0;
+
+    logicalDisk.GetString("Description", r["type"]);
```

Comment:
This is a nice improvement!  Can you please update the schema docs to indicate the known expected values for type field?  On my win10 VM they are: 'Logical Fixed Disk', 'Network Connection', 'CD-ROM Disk'.